linkml/generators/common/ifabsent_processor.py

```diff
@@ -284,3 +293,27 @@ def _uri_for(self, s: str) -> str:
     def _strval(self, txt: str) -> str:
         txt = str(txt).replace('"', '\\"')
         return f'"{txt}"'
+
+    def _map_uri_special_case(self, default_value: str, slot: SlotDefinition, cls: ClassDefinition) -> str:
+        if default_value == "class_uri":
+            return f'"{self.schema_view.get_uri(cls, expand=True)}"'
+        elif default_value == "slot_uri":
+            return f'"{self.schema_view.get_uri(slot, expand=True)}"'
+        else:
+            raise ValueError(
+                f"Default value must be one of the URI special cases: {self.URI_SPECIAL_CASES}. Got: {default_value}"
+            )
+
+    def _map_curie_special_case(self, default_value: str, slot: SlotDefinition, cls: ClassDefinition) -> str:
+        if default_value == "class_curie":
+            return f'"{self.schema_view.get_uri(cls, expand=False)}"'
+        elif default_value == "slot_curie":
+            return f'"{self.schema_view.get_uri(slot, expand=False)}"'
+        else:
+            raise ValueError(
+                f"Default value must be one of the curie special cases: {self.CURIE_SPECIAL_CASES}. "
+                f"Got: {default_value}"
+            )
+
+    def _map_default_range_special_case(self, default_value: str, slot: SlotDefinition, cls: ClassDefinition) -> str:
+        return f'"{self.schema_view.schema.default_range}"'
```

Comment:
What is the reason for having so many useless arguments in this method? Signature consistency? It is not an abstract method that might get different functionality that requires those arguments.

Why not simply?
```suggestion
    def _map_default_range_special_case(self) -> str:
        return f'"{self.schema_view.schema.default_range}"'
```

Reply:
Api consistency is the answer - some other generator may need the slot and class to know how to render that value, even if it's not very likely since it should only change on a schema level. So eg. In the shacl generator how it needs to wrap the value in some RDF literal, that could depend on the range in the slot, and who knows maybe the class has some additional rules that condition it.